lib/Dialect/TritonGPU/IR/LinearLayoutConversions.cpp

```diff
@@ -793,4 +793,70 @@ LinearLayout chooseShemLayoutForRegToRegConversion(
       {{kOffset, totalOffsets}, {kIteration, totalIters}, {kBlock, 1}});
 }
 
+namespace {
+
+// TODO (Keren): Currently, we have more restrictions than necessary when using
+// stmatrix.  These restrictions are retained from legacy code, and we could
+// relax some of them in the future.
+bool canUseStMatrix(RankedTensorType tensorTy, ArrayRef<unsigned> repShape,
+                    ArrayRef<unsigned> paddedRepShape,
+                    ArrayRef<unsigned> order) {
+  auto mmaLayout =
+      mlir::dyn_cast<NvidiaMmaEncodingAttr>(tensorTy.getEncoding());
+  if (!mmaLayout || !mmaLayout.isHopper())
+    return false;
+  if (tensorTy.getElementType().getIntOrFloatBitWidth() != 16)
+    return false;
+  if (order[0] != 1)
+    return false;
+
+  auto tensorShape = tensorTy.getShape();
+  if (tensorShape.size() != 2)
+    return false;
+  auto numIterations = ceil<unsigned>(tensorShape[1], repShape[1]) *
+                       ceil<unsigned>(tensorShape[0], repShape[0]);
+  if (numIterations > 1)
+    return false;
+  if (paddedRepShape[1] % 8 != 0)
+    return false;
+  return true;
+}
+
+} // anonymous namespace
+
+std::optional<LinearLayout> chooseStMatrixLayoutForRegToRegConversion(
```

Comment:
could you document the meaning of the result here? It is not super clear to me as the usage suggests this is the shared memory layout but from the implementation that doesn't seem to be.

Reply:
Added
```
// If `stmatrix` cannot be used for the given tensor encoding, this function
// returns `std::nullopt`.
```

Hope it's clearer